src/script/builder/tir/prim_func_frame.cc

```diff
@@ -80,17 +107,38 @@ tvm::tir::Buffer Arg(String name, tvm::tir::Buffer buffer) {
 
 void FuncName(String name) {
   PrimFuncFrame frame = Builder::Current()->FindFrame<PrimFuncFrame>().value();
+  Optional<BlockFrame> res = Builder::Current()->GetLastFrame<BlockFrame>();
+  if (!(res.defined() && frame->root_block_frame.same_as(res.value()))) {
+    LOG(FATAL) << "Prim func name should be declared before root block";
+  }
```

Comment:
How about we have a common method in `prim_func_frame.cc` which finds the PrimFuncFrame and do this check? Like:

```C++
PrimFuncFrame FindPrimFuncFrame(const String& method) {
  Builder builder = Builder::Current();
  if (Optional<PrimFuncFrame> prim_func_frame = builder->FindFrame<PrimFuncFrame>()) {
    if (Optional<BlockFrame> block_frame = builder->GetLastFrame<BlockFrame>()) {
      if (prim_func_frame.value()->root_block_frame.get() == block_frame.get()) {
        return prim_func_frame.value();
      }
    }
  } else {
    LOG(FATAL) << "ValueError: PrimFunc frame not find. Please ensure '" << method << "' is called under T.prim_func()";
  }
  LOG(FATAL) << "ValueError: '" << method << "' must be called immediately under T.prim_func()";
  throw;
}
```